components/booking/CalendarCell.tsx

```diff
@@ -0,0 +1,171 @@
+import { useState } from 'react'
+import {
+  Modal,
+  ModalOverlay,
+  ModalHeader,
+  ModalBody,
+  ModalContent,
+  ModalCloseButton,
+  Button,
+  ModalFooter,
+  Box,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import { motion } from 'framer-motion'
+import { format } from 'date-fns'
+import { getVenueFromId } from '../../utils'
+import { useAllVenues } from '../../hooks/useAllVenues'
+
+interface CellProps {
+  text: string
+  isExpanded: boolean
+  isSelected: boolean
+  onClick?: () => void
+  bookings: Array<BookingDataDisplay>
+}
+
+const spring = {
+  type: 'spring',
+  damping: 50,
+  stiffness: 300,
+}
+
+const CalendarCell: React.FC<CellProps> = ({ text, isExpanded, isSelected, onClick, bookings }) => {
+  const [allVenues, isLoadingVenues] = useAllVenues()
+  const [modalOpen, setModalOpen] = useState(false)
+
+  const toggleBookingModal = (open: any) => {
+    setModalOpen(open)
+  }
```

Comment:
Can we use `useDisclosure()` for this?

Reply:
Subject: [PATCH] Bugfix
---
Index: components/booking/CalendarCell.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/components/booking/CalendarCell.tsx b/components/booking/CalendarCell.tsx
--- a/components/booking/CalendarCell.tsx	(revision 957f1697edc3f655229e10207cff73500fdf8fed)
+++ b/components/booking/CalendarCell.tsx	(date 1688283011180)
@@ -1,4 +1,3 @@
-import { useState } from 'react'
 import {
   Modal,
   ModalOverlay,
@@ -6,11 +5,10 @@
   ModalBody,
   ModalContent,
   ModalCloseButton,
-  Button,
-  ModalFooter,
   Box,
   Text,
   VStack,
+  useDisclosure,
 } from '@chakra-ui/react'
 import { motion } from 'framer-motion'
 import { format } from 'date-fns'
@@ -33,11 +31,7 @@
 
 const CalendarCell: React.FC<CellProps> = ({ text, isExpanded, isSelected, onClick, bookings }) => {
   const [allVenues, isLoadingVenues] = useAllVenues()
-  const [modalOpen, setModalOpen] = useState(false)
-
-  const toggleBookingModal = (open: any) => {
-    setModalOpen(open)
-  }
+  const { isOpen, onOpen, onClose } = useDisclosure()
 
   const list = {
     visible: {
@@ -109,7 +103,7 @@
                     textAlign='left'
                     cursor='pointer'
                     _hover={{ textDecoration: 'underline' }}
-                    onClick={() => toggleBookingModal(true)}
+                    onClick={onOpen}
                   >
                     {bookings.length - 3} more
                   </Text>
@@ -140,7 +134,7 @@
           layoutId='underline'
         />
       )}
-      <Modal isOpen={modalOpen} onClose={() => toggleBookingModal(false)}>
+      <Modal isOpen={isOpen} onClose={onClose}>
         <ModalOverlay />
         <ModalContent>
           <ModalHeader>Bookings</ModalHeader>
@@ -156,12 +150,6 @@
               )
             })}
           </ModalBody>
-
-          <ModalFooter>
-            <Button colorScheme='blue' mr={3} onClick={() => toggleBookingModal(false)}>
-              Close
-            </Button>
-          </ModalFooter>
         </ModalContent>
       </Modal>
     </motion.div>

Reply:
Here's a patchfile for this
[useDisclosure.patch.txt](https://github.com/usdevs/usc-website-hackathon-frontend/files/11928925/conrad-dev.patch.txt)